310 fallback service provider (#315)
egil/bUnit: docs/site/docs/providing-input/inject-services-into-components.md

> [!NOTE]
> The `AddSingleton()` method is only available on the `Services` collection if you import the `Microsoft.Extensions.DependencyInjection` type.

> ##  Custom Service Provider

Reviewer: ```suggestion
##  Fallback Service Provider
```

> Here is an example on how to implement and use a fallback service provider. This service provider gets used when the default service provider returns null for the requested type.

Reviewer: I think we need to start this section with a short sentence that motivates the usage of a fallback service provider. Explain the main use case is to use a mocking framework to automatically create mocks of requested services.

> [!NOTE]
> The `AddSingleton()` method is only available on the `Services` collection if you import the `Microsoft.Extensions.DependencyInjection` type.

##  Fallback Service Provider

> A fallback service provider can be registered with the built-in `TestServiceProvider`. This enables a few interesting use-cases, such as using an alternative IoC container, as long as they implement the `IServiceProvider` interface, or automatically create mock services for your Blazor components. The latter can be achieved by using a combination of [AutoFixture](https://github.com/AutoFixture/AutoFixture) and your favorite mocking framework, e.g. Moq, NSubsitute, or Telerik JustMock.

Reviewer 2: I'd question need for hyphen in 'use-cases', and would drop the comma before 'automatically create mock service'. Also 'it implements' rather than 'they implement', as you're talking about using an alternative container (singular), not containers (plural). Hence: "This enables a few interesting use cases, such as using an alternative IoC container, as long as it implements the `IServiceProvider` interface or automatically creates mock services for your Blazor components." (Or should it actually be plural - they implement/automatically create - since you're talking about use cases? It depends whether the reader should see the entire pair of clauses 'as long as they ... Blazor components' as relating primarily to the first clause ('This enables ... use cases') or the following clause ('such as ... container'). Am I making this clear?(!)

> A fallback service provider can be registered with the built-in `TestServiceProvider`. This enables a few interesting use-cases, such as using an alternative IoC container, as long as they implement the `IServiceProvider` interface, or automatically create mock services for your Blazor components. The latter can be achieved by using a combination of [AutoFixture](https://github.com/AutoFixture/AutoFixture) and your favorite mocking framework, e.g. Moq, NSubsitute, or Telerik JustMock.

Reviewer 2: ```suggestion
A fallback service provider can be registered with the built-in `TestServiceProvider`. This enables a few interesting use cases, such as using an alternative IoC container (which should implement the `IServiceProvider` interface), or automatically creating mock services for your Blazor components. The latter can be achieved by using a combination of [AutoFixture](https://github.com/AutoFixture/AutoFixture) and your favorite mocking framework, e.g. Moq, NSubsitute, or Telerik JustMock.
```
